### ISSUE TITLE
numpy 1.10.4 + mkl on anaconda leads to test failure

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#177]
With the master branch - b1e905e5fdbd1579d0fe8757584619fed27fde43 - I have just seen the following test failure (from manually running `sherpa/utils/tests/test_utils.py`, but you also get it from `python setup.py test`).

```
======================================================================
FAIL: test_parallel_map (sherpa.utils.tests.test_utils.test_utils)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/djburke/sherpa/sherpa-test/sherpa/utils/tests/test_utils.py", line 214, in test_parallel_map
    self.assert_((numpy.asarray(result) == numpy.asarray(pararesult)).all())
AssertionError: False is not true
```

This is in a new conda environment - I noticed it updated numpy, so maybe that's the cause.

```
% conda list numpy
# packages in environment at /home/djburke/miniconda2/envs/sherpa-test:
#
numpy                     1.10.4                   py27_0  
```

**Comment by DougBurke** [comment]
If I downgrade to `numpy 1.10.2` then the tests pass, so it looks like something in either `numpy 1.10.4`, or the associated `mkl` package, is causing the problem:

```
conda update numpy
Fetching package metadata: ......
Solving package specifications: ................
Package plan for installation in environment /home/djburke/miniconda2/envs/sherpa-test2:

The following NEW packages will be INSTALLED:

    mkl:   11.3.1-0     

The following packages will be UPDATED:

    numpy: 1.10.2-py27_0 --> 1.10.4-py27_0

Proceed ([y]/n)? y

Unlinking packages ...
[      COMPLETE      ]|###################################################| 100%
Linking packages ...
[      COMPLETE      ]|###################################################| 100%
```

**Comment by DougBurke** [comment]
It appears to be a tolerance issue - changing to use `numpy.testing.assert_allclose` seems to fix this problem. We should really review this particular test as to see if it is a sensible choice.

**Comment by DougBurke** [comment]
I have two different approaches to work around this issue:

* #178 which compares the arrays using a tolerance rather than using equality 

* #180 change the `.travis.yml` file to force numpy to be '< 1.10.4'

My suggestion would be for #180, as a very short-term fix, then we can decide if #178 is a sensible fix or if the test should be changed.

**Comment by olaurino** [comment]
There is a third option, which is probably the one I like the best for the time being, assuming it works: opt out of `mkl` and keep installing the latest 1.10 version available.

I'll try putting together a PR after lunch.

#178 should be tested on macs. I would like to have a better sense of how different the results are with mkl packages, and there is a chance (it may be just wishful thinking) that mkl-enabled packages may work-around the numerical differences seen on Linux vs OSX.

**Comment by DougBurke** [comment]
I've merged #181 which should work around this issue while we have time to review the test and why it failed with MKL.

**Comment by olaurino** [comment]
Fixed by #178
